src/var/mash.jl

```diff
@@ -0,0 +1,52 @@
+# MinHash
+# ========
+#
+# Functions to get MASH distance for MinHash sketches
+#
+# see DOI: 10.1186/s13059-016-0997-x
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+function jaccarddistance(sketch1::MinHashSketch, sketch2::MinHashSketch)
+    sketch1.kmersize == sketch2.kmersize || error("sketches must have same kmer length")
+    length(sketch1) == length(sketch2) || error("sketches must be the same size")
+
+    matches = 0
+    sketchlen = length(sketch1)
+    i = 1
+    j = 1
+
+    while i <= sketchlen && j <= sketchlen
+        if sketch1.sketch[i] == sketch2.sketch[j]
+            matches += 1
+            i += 1
+            j += 1
+        elseif sketch1.sketch[i] < sketch2.sketch[j]
+            while sketch1.sketch[i] < sketch2.sketch[j]
+                i += 1
+            end
+        elseif sketch2.sketch[j] < sketch1.sketch[i]
+            while sketch2.sketch[j] < sketch1.sketch[i]
+                j += 1
+            end
+        end
+    end
+    matches == sketchlen ? (return 1.0) : return matches / (2 * sketchlen - matches)
```

Comment:
I agree, though I thought using the ternary operator is more "standard" in julia.

It's worth noting that this syntax is currently in a bunch of places around the Bio.jl code. Admittedly in most cases, they're in quite short and clear places, but in others, not so much eg in banded_needleman_wunch.jl:

```julia
@update2_nextcol (i == m ? end_gap_init_a : middle_gap_init_a) (i == m ? end_gap_extend_a : middle_gap_extend_a)
```

As someone coming from python, I was really confused by this syntax initially, and it's not easy to find info about it. I fully support using if-else in most circumstances, but in that case we should probably be more consistent (and maybe add something to CONTRIBUTING.md if we're breaking from standard julia style).